### PR TITLE
feat(assembly reference resolving): Support reference aliases

### LINKED ISSUE #2601: Exception when using PackageReference aliases
The bug causes a runtime exception when a source project contains a PackageReference alias, such as:

`<PackageReference Include="Dapper" Version="2.0.143" Aliases="ORM" />`

**Logs**
[logs.zip](https://github.com/stryker-mutator/stryker-net/files/11998240/logs.zip)

**Expected behavior**
No runtime errors

**Desktop**
 - OS: Windows
 - Type of project: core
 - Framework Version: net6.0
 - Stryker Version 3.9.2

**Additional context**
Nothing in particular. It seems quite easy to reproduce by adding a nuget reference to any old package and adding an alias.

### PR DESCRIPTION
**Pull request by dupdob**
add support for alias(es) for assembly references.
Note that I failed to find documentation about having multiple aliases for the same reference, so support for those is provisional only.
Fixes #2601 
minor: fix problem with release mode.

### CODE REVIEW

File: src/Stryker.CLI/Stryker.CLI/runtimeconfig.template.json [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-  "rollForwardOnNoCandidateFx": 2
+  //"rollForwardOnNoCandidateFx": 2
```
Comment (rouke-broersma) [inline src/Stryker.CLI/Stryker.CLI/runtimeconfig.template.json:2]: Accidental commit?

Comment (dupdob) [reply src/Stryker.CLI/Stryker.CLI/runtimeconfig.template.json:2]: yes, sorry about that. It got embarked with the merge, I guess.
By the way, the problem is that VS 17.7 does add RollForward on its own, so it conflicts with this. So far, I failed to find anything to either control this setting or ensure a correct result

Comment (psfinaki) [reply src/Stryker.CLI/Stryker.CLI/runtimeconfig.template.json]: Yeah btw, so this is not only me, had to do a similar hack locally. `rollForward` also works...

**Review by psfinaki — APPROVED** [review]

**Comment by SteveDunn** [comment]
Hi - any idea when this will be merged and released? I'm happy to help if I can, in achieving this.

**Comment by rouke-broersma** [comment]
> Hi - any idea when this will be merged and released? I'm happy to help if I can, in achieving this.

Next week probably 🤷

**Pull request merged**